storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManager.java

```diff
@@ -76,6 +78,11 @@ public class TopicBasedRemoteLogMetadataManager implements RemoteLogMetadataMana
     private final boolean startConsumerThread;
 
     private Thread initializationThread;
+
+    private static final long SHUTDOWN_TIMEOUT_SECONDS = 60L;
+
+    private CountDownLatch initializeLatch;
```

Comment:
Just a suggestion, but I think you could simplify tracking the initialization flow by folding  `initializeLatch` & `initialized` into a `CompletableFuture` (which could also handle the case when close is called while the initialization is still in progress).

Reply:
Thanks for the tip! Definitely can do this, I was just mimicking the style in the rest of the Kafka codebase. Will see what the reviewers respond with.